### PR TITLE
[ENG-807] List view improvements

### PR DESCRIPTION
**Pull request by niikeec**

### CODE REVIEW

File: interface/hooks/useIsTextTruncated.ts [hunk 1]
```diff
@@ -1,29 +1,26 @@
...
+
+	useCallbackToWatchResize(
+		() => setIsTruncated(determineIsTruncated()),
+		[determineIsTruncated],
+		element
+	);
+
+	useEffect(() => {
+		setIsTruncated(determineIsTruncated());
+	}, [element, determineIsTruncated, text]);
```
Comment (Brendonovich) [inline interface/hooks/useIsTextTruncated.ts:23]: Ik this was here before you changed it but could the state + callback + effect be combined into a memo?

**Review by Brendonovich — CHANGES_REQUESTED** [review]
all da contexts

File: interface/hooks/useIsTextTruncated.ts (hunk 1 above)
Comment (niikeec) [reply interface/hooks/useIsTextTruncated.ts:23]: If you prefer this
```ts
export const useIsTextTruncated = (element: RefObject<HTMLElement>, text: string | null) => {
    const { width } = useResizeObserver({ ref: element });

    return useMemo(() => {
        if (!element.current) return false;
        return element.current.scrollWidth > element.current.clientWidth;
    }, [element, width, text]);
};
```

Comment (Brendonovich) [reply interface/hooks/useIsTextTruncated.ts:23]: Ye that should do it

**Review by Brendonovich — APPROVED** [review]

**Pull request merged**